### PR TITLE
checker: migrate test framework to testify

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

Issue Number: Ref #4813.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/testutil/operator_check.go [hunk 1]
```diff
@@ -141,3 +142,62 @@ func CheckTransferPeerWithLeaderTransferFrom(c *check.C, op *operator.Operator,
...
+	re.Equal(sourceID, steps[2].(operator.RemovePeer).FromStore)
+	kind |= operator.OpRegion
+	re.Equal(kind, op.Kind()&kind)
+}
+
+// CheckSteps checks if the operator matches the given steps.
+func CheckSteps(re *require.Assertions, op *operator.Operator, steps []operator.OpStep) {
+	re.NotEqual(0, op.Kind()&operator.OpMerge)
+	re.NotNil(steps)
+	re.Equal(len(steps), op.Len())
```
Comment (JmPotato) [inline pkg/testutil/operator_check.go:185]:
```suggestion
	re.Len(steps, op.Len())
```

File: server/schedule/checker/joint_state_checker_test.go [hunk 2]
```diff
@@ -131,38 +114,38 @@ func (s *testJointStateCheckerSuite) TestLeaveJointState(c *C) {
...
 		return
 	}
 
-	c.Assert(op, NotNil)
-	c.Assert(op.Desc(), Equals, "leave-joint-state")
+	re.NotNil(op)
+	re.Equal("leave-joint-state", op.Desc())
 
-	c.Assert(op.Len(), Equals, len(steps))
+	re.Equal(len(steps), op.Len())
```
Comment (JmPotato) [inline server/schedule/checker/joint_state_checker_test.go:130]: Ditto.

File: server/schedule/checker/merge_checker_test.go [hunk 3]
```diff
@@ -36,163 +36,159 @@ import (
...
 	ctx     context.Context
 	cancel  context.CancelFunc
 	cluster *mockcluster.Cluster
 	mc      *MergeChecker
 	regions []*core.RegionInfo
 }
 
-func (s *testMergeCheckerSuite) SetUpSuite(c *C) {
-	s.ctx, s.cancel = context.WithCancel(context.Background())
+func TestMergeCheckerSuite(t *testing.T) {
```
Comment (JmPotato) [inline server/schedule/checker/merge_checker_test.go:52]:
```suggestion
func TestMergeCheckerTestSuite(t *testing.T) {
```

**Review by JmPotato — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**